core/domain/stats_jobs_one_off.py

```diff
@@ -341,9 +341,23 @@ def reduce(exp_id, stringified_values):
                             'state_name']] = (
                                 stats_domain.StateStats.create_default())
                     elif change_dict['cmd'] == exp_domain.CMD_DELETE_STATE:
+                        if change_dict['state_name'] not in state_stats_mapping:
+                            yield (
+                                'State name %s not in state stats mapping of '
+                                'exploration with ID %s and version %s' % (
+                                    change_dict['state_name'], exp_id, version))
+                            continue
                         state_stats_mapping.pop(
                             change_dict['state_name'])
                     elif change_dict['cmd'] == exp_domain.CMD_RENAME_STATE:
+                        if change_dict['old_state_name'] not in (
+                                state_stats_mapping):
+                            yield (
+                                'State name %s not in state stats mapping of '
```

Comment:
Good idea to catch this here too, but don't you want to distinguish the error messages for the two cases? Otherwise you won't know what triggered it.

Also, I'm not sure if we should just continue along the changelist if we find an error. Shouldn't we just ... stop, and not process that exploration? Otherwise there may be further errors down the line, e.g. if the new state name subsequently doesn't appear in the changelist.

Reply:
Makes sense. Done.